src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2447 @@
+            loss=loss_val,
+        )
+
+
+@dataclass
+class PatchTSMixerForRegressionOutput(ModelOutput):
+    """
+    Output type of [`PatchTSMixerForRegressionOutput`].
+
+    Args:
+        prediction_logits (`torch.FloatTensor` of shape `(batch_size, num_targets)`):
+            Prediction output from the regression head.
+        last_hidden_state (`torch.FloatTensor` of shape `(batch_size, num_input_channels, num_patches, num_features)`):
+            Backbone embeddings before passing through the head.
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*):
+            Hidden-states of the model at the output of each layer plus the optional initial embedding outputs.
+        loss (*optional*, returned when `y` is provided, `torch.FloatTensor` of shape `()`):
+            Total loss.
+    """
+
+    prediction_logits: torch.FloatTensor = None
+    last_hidden_state: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    loss: Optional[torch.FloatTensor] = None
+
+
+class PatchTSMixerForRegression(PatchTSMixerPreTrainedModel):
+    r"""
+    `PatchTSMixer` for regression application.
+
+    Args:
+        config (`PatchTSMixerConfig`, *required*):
+            Configuration.
+
+    Returns:
+        `None`.
+    """
+
+    def __init__(self, config: PatchTSMixerConfig):
+        super().__init__(config)
+
+        self.model = PatchTSMixerModel(config)
+        self.config = config
+        if config.loss == "mse":
+            self.distribution_output = None
+        else:
+            if config.distribution_output == "student_t":
+                self.distribution_output = StudentTOutput(dim=config.num_targets)
+            elif config.distribution_output == "normal":
+                self.distribution_output = NormalOutput(dim=config.num_targets)
+            elif config.distribution_output == "negative_binomial":
+                self.distribution_output = NegativeBinomialOutput(dim=config.num_targets)
+            else:
+                raise ValueError(f"Unknown distribution output {config.distribution_output}")
+
+        if config.scaling in ["std", "mean", True]:
+            if config.mode == "flatten":
+                raise ValueError("Scaling is not supported for classification task when mode == flatten")
+            self.inject_scale = InjectScalerStatistics4D(
+                num_features=config.num_features, num_patches=config.num_patches
+            )
+        else:
+            self.inject_scale = None
+
+        self.head = PatchTSMixerLinearHead(
+            config=config,
+            distribution_output=self.distribution_output,
+        )
+
+        # Initialize weights and apply final processing
+        if config.post_init:
+            self.post_init()
+
+    @add_start_docstrings_to_model_forward(PATCHTSMIXER_INPUTS_DOCSTRING)
+    # @replace_return_docstrings(output_type=PatchTSMixerForRegressionOutput, config_class=_CONFIG_FOR_DOC)
```

Comment:
to remove?

Reply:
https://github.com/huggingface/transformers/pull/26247/commits/ec6ea20945e4fa817f627c49844882d17084ecc8